tracer/test/Datadog.Trace.IntegrationTests/StatsTests.cs

```diff
@@ -568,7 +568,7 @@ void AssertTraces(IReadOnlyList<MockSpan> payload, bool expectStats)
                 // occurrence would be considered "rare". However, the RareSampler
                 // only runs after the PrioritySampler, so assert the metric "_dd.rare"
                 // based on the SamplingPriority
-                bool expectRareMetric = expectStats && payload[0].GetMetric(Metrics.SamplingPriority) <= 0;
+                bool expectRareMetric = expectStats && payload[0].GetMetric(Metrics.SamplingPriority) != (int)SamplingPriority.UserReject;
```

Comment:
Why `payload[0]`? Sampling priority is usually added to the local root span (if present), which is usually the last span in the chunk. With partial flushing the local root may not even be in this chunk.

Unless you know that `payload.Count == 1`, you may need to check every span until you find the sampling priority to be safe.

Reply:
That's a good question. I didn't write that test (I just updated the condition), I need to review the logic